sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystTypeConverters.scala

```diff
@@ -457,7 +457,9 @@ object CatalystTypeConverters {
     case d: JavaBigDecimal => new DecimalConverter(DecimalType(d.precision, d.scale)).toCatalyst(d)
     case seq: Seq[Any] => new GenericArrayData(seq.map(convertToCatalyst).toArray)
     case r: Row => InternalRow(r.toSeq.map(convertToCatalyst): _*)
-    case arr: Array[Any] => new GenericArrayData(arr.map(convertToCatalyst))
+    case arr: Array[Byte] => arr
+    case arr: Array[Char] => StringConverter.toCatalyst(arr)
+    case arr: Array[_] => new GenericArrayData(arr.map(convertToCatalyst))
```

Comment:
How does this PR fixes `Literal(Array(1, 2, 3)) == Literal.create(Array(1, 2, 3), ArrayType(IntegerType))` case?

Reply:
`Array[Any]` doesn't match the `Array(1, 2, 3)`. This PR use the `Array[_]` to match it.

Reply:
It already works without this change.

```scala
scala> Literal(Array(1, 2, 3))
res1: org.apache.spark.sql.catalyst.expressions.Literal = [1,2,3]
```

Reply:
Seems missed a param `ArrayType(IntegerType)`, they use the different code path.

Reply:
Oh, gotya.